php-practice/section12.php

```diff
@@ -159,3 +214,54 @@ class TwelveSidedDie extends Dice {
  * HINT: What shared methods or properties need to be in the parent HatOfNames class?
  * HINT: What visibility do these shared methods and properties need? (public, private, or protected?)
  */
+
+abstract class HatOfNames {
+  const MODE_RETAIN_WINNER = 1;
+  const MODE_NEW_WINNER_EVERY_TIME = 2;
+
+  protected $names = [];
+  protected $winner;
+  protected $mode;
```

Comment:
So good job on this, although it's not quite right.

The idea of "mode" should be removed because it's no longer necessary. Instead of specifying a mode, the developer just has to pick the correct kind of class. (Instead of specifying the "locked" mode, you should just use the "locked" type of class)

As it stands, it's problematic. What if I make a `LockedHatOfNames` and pass in the `self::MODE_NEW_WINNER_EVERY_TIME` mode to the constructor? What will `pickWinner()` return?